tests/pytorch/r2.1/sd-model.libsonnet

```diff
@@ -0,0 +1,106 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+local experimental = import '../experimental.libsonnet';
+local common = import 'common.libsonnet';
+local timeouts = import 'templates/timeouts.libsonnet';
+local tpus = import 'templates/tpus.libsonnet';
+local utils = import 'templates/utils.libsonnet';
+
+{
+  local sd_model = self.sd_model,
+  sd_model:: common.PyTorchTest {
+    local config = self,
+    modelName: 'lightning-sd-model',
+    paramsOverride:: {
+      scriptPath: 'stable-diffusion/main_tpu.py',
+      trainCommand: [
+        'python3',
+        self.scriptPath,
+        '--train',
+        '--no-test',
+        '--base=stable-diffusion/configs/latent-diffusion/cin-ldm-vq-f8-ss.yaml',
+        '--',
+        'data.params.batch_size=32',
+        'lightning.trainer.max_epochs=5',
+        'model.params.first_stage_config.params.ckpt_path=stable-diffusion/models/first_stage_models/vq-f8/model.ckpt',
+        'lightning.trainer.enable_checkpointing=False',
+        'lightning.strategy.sync_module_states=False',
+      ],
+    },
+    command: self.paramsOverride.trainCommand,
+  },
+  local pjrt = self.pjrt,
+  pjrt:: common.PyTorchTpuVmMixin {
+    modelName+: '-pjrt',
+    tpuSettings+: {
+      tpuVmExtraSetup: |||
+        git clone https://github.com/pytorch-tpu/stable-diffusion.git
+        cd stable-diffusion
+        pip install transformers==4.19.2 diffusers invisible-watermark
+        pip install -e .
+        pip install torchmetrics==0.7.0
+        pip install https://github.com/Lightning-AI/lightning/archive/refs/heads/master.zip -U
+        pip install lmdb einops omegaconf
+        pip install taming-transformers clip kornia==0.6 albumentations==0.4.3
+        pip install starlette==0.27.0 && pip install tensorboard
+        sudo apt-get update -y && sudo apt-get install libgl1 -y
+        # wget -nv https://github.com/CompVis/taming-transformers/blob/master/taming/modules/vqvae/quantize.py
+        pip install -e git+https://github.com/CompVis/taming-transformers.git@master#egg=taming-transformers
+        echo w | pip install -e git+https://github.com/openai/CLIP.git@main#egg=clip
+        # mv quantize.py ~/.local/lib/python3.8/site-packages/taming/modules/vqvae/
+
+        # taming-transformers and CLIP override existing torch and torchvision so we need to reinstall
+        # TODO change back to torch2.1 once pytorch released torch2.1
+        # pip uninstall -y torch torchvision
+        # pip install --user \
+        #   https://storage.googleapis.com/pytorch-xla-releases/wheels/xrt/tpuvm/torch-nightly-cp310-cp310-linux_x86_64.whl \
```

Comment:
We'll leave those for now just in case for reference. They are commented out for now anyways.

Reply:
Please refer to https://github.com/GoogleCloudPlatform/ml-testing-accelerators/pull/977#discussion_r1313587218. I'm not sure why in this PR these lines are commented out but they need to be added back.

Reply:
Addressed in the comment thread above.